### PR TITLE
Upgrading envoy to v1.25.1 and related dependencies

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Fixes: https://github.com/wso2/apk/issues/737

## Related PRs
https://github.com/wso2/product-microgateway/pull/3216
https://github.com/wso2/product-microgateway/pull/3225
https://github.com/wso2/product-microgateway/pull/3257

## Tested Images
sampathrajapakse/router:0.0.1-test3
sampathrajapakse/enforcer:0.0.1-test3
sampathrajapakse/adapter:0.0.1-test3

### CODE REVIEW

File: adapter/go.mod [hunk 1]
```diff
@@ -3,21 +3,21 @@ module github.com/wso2/apk/adapter
 go 1.19
 
 require (
-	github.com/envoyproxy/go-control-plane v0.10.3
+	github.com/envoyproxy/go-control-plane v0.10.3-0.20230119052701-0ebb8a16ea22
```
Comment (AmaliMatharaarachchi) [inline adapter/go.mod:6]: shall we update to latest go control plane version?

Comment (AmaliMatharaarachchi) [reply adapter/go.mod:6]: 0.11.0

File: libs.versions.toml [hunk 2]
```diff
@@ -140,7 +140,7 @@ commons-logging = "1.1.1"
 commons-pool = "1.5.6.wso2v1"
 commons-validator = "1.7"
 cxf = "3.5.4"
-envoyproxy = "0.1.28"
+envoyproxy = "1.0.36"
```
Comment (AmaliMatharaarachchi) [inline libs.versions.toml:143]: latest is 1.0.37

File: adapter/go.mod (hunk 1 above)
Comment (CrowleyRajapakse) [reply adapter/go.mod]: fixed

File: libs.versions.toml (hunk 2 above)
Comment (CrowleyRajapakse) [reply libs.versions.toml]: fixed

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**